pkg/controller/pod_control.go

```diff
@@ -351,3 +358,14 @@ func (c *FakePodControl) UpdatePod(_ runtime.Object, pod *corev1.Pod) (*corev1.P
 }
 
 var _ PodControlInterface = &FakePodControl{}
+
+func PdNameWithPodName(tc *v1alpha1.TidbCluster, pod *corev1.Pod) string {
+	name := pod.Name
+	if tc.IsMultiClusterEnabled() {
+		name = fmt.Sprintf("%s.%s-pd-peer.%s.svc.%s", pod.Name, tc.GetName(), tc.Namespace, tc.Spec.ClusterDomain)
+		if _, exist := tc.Status.PD.Members[name]; !exist {
+			name = pod.Name
```

Comment:
Need to check `tc.Status.PD.Members[pod.Name]` here?
What if both names do not exist in the `Status.PD.Members`?